app/src/main/java/org/dhis2/usescases/enrollment/EnrollmentActivity.kt

```diff
@@ -233,69 +238,73 @@ class EnrollmentActivity : ActivityGlobalAbstract(), EnrollmentView {
     }
 
     override fun goBack() {
-        if (presenter.dataIntegrityCheck(adapter.mandatoryOk(), adapter.hasError())) {
-            onBackPressed()
-        }
+        onBackPressed()
     }
 
     override fun showMissingMandatoryFieldsMessage() {
         showInfoDialog(
-                getString(R.string.unable_to_complete),
-                getString(R.string.missing_mandatory_fields)
+            getString(R.string.unable_to_complete),
+            getString(R.string.missing_mandatory_fields)
         )
     }
 
     override fun showErrorFieldsMessage() {
         showInfoDialog(
-                getString(R.string.unable_to_complete),
-                getString(R.string.field_errors)
+            getString(R.string.unable_to_complete),
+            getString(R.string.field_errors)
         )
     }
 
     override fun onBackPressed() {
-        if (mode == EnrollmentMode.CHECK) super.onBackPressed() else showDeleteDialog()
+        if (mode == EnrollmentMode.CHECK) {
+            if (presenter.dataIntegrityCheck(adapter.mandatoryOk(), adapter.hasError())) {
+                super.onBackPressed()
+            }
+        } else {
+            showDeleteDialog()
+        }
```

Comment:
To me is okay :+1: . If you want, I would have done something.
`override fun onBackPressed() {
  presenter.onBackPressed(EnrollmentMode.Check)
}`

And then move as much logic to presenter, then unit test it and forget about this bug forever :)